synapse/tests/test_lib_ast.py

```diff
@@ -0,0 +1,17 @@
+import synapse.common as s_common
+import synapse.tests.utils as s_test
+
+class AstTest(s_test.SynTest):
+
+    async def test_ast_subq_vars(self):
+
+        async with self.getTestCore() as core:
+
+            q = '''
+                $loc=newp
+                [ test:comp=(10, lulz) ]
+                { -> test:int [ :loc=haha ] $loc=:loc }
+                $lib.print($loc)
+            '''
```

Comment:
The following query currently works without the change in ast.py, since the ``$loc`` assignment assigns the value to the runtime, and the value is then pulled from the path object, which reaches through the runtime on a miss.

```
 q = '''
                [ test:comp=(10, lulz) ]
                { -> test:int [ :loc=haha ] $loc=:loc }
                $lib.print($loc)
            '''
```